lib/devise.rb

```diff
@@ -280,6 +341,32 @@ module Strategies
   # a fresh initializer with all configuration values.
   def self.setup
     yield self
+
+    warn_default_config_changed(:email_regexp, '/\A[^@\s]+@([^@\s]+\.)+[^@\W]+\z/', '/\A[^@\s]+@[^@\s]+\z/')
+    warn_default_config_changed(:reconfirmable, 'false', 'true')
+    warn_default_config_changed(:sign_out_via, ':get', ':delete')
+    warn_default_config_changed(:skip_session_storage, '[]', '[:http_auth]')
+    warn_default_config_changed(:strip_whitespace_keys, '[]', '[:email]')
+  end
+
+  def self.warn_default_config_changed(config, current_default, new_default)
+    unless config.in?(explicit_set_configs)
+      warn = <<-MESSAGE.strip_heredoc
+        [Devise] config.#{config} will have new default in the Devise 4.1
+        To keep the current behavior please set in your devise.rb initializer file:
+
+          Devise.setup do |#{config}|
+            config.#{config} = #{current_default}
+          end
+
+        If you want to use the new default:
+
+          Devise.setup do |#{config}|
+            config.#{config} = #{new_default}
+          end
+      MESSAGE
+      ActiveSupport::Deprecation.warn(warn) unless :email_regexp.in?(explicit_set_configs)
```

Comment:
We don't move the configs to a constant because the main idea was only warn people that are using the default configurations.